### PR TITLE
Don't use a Singleton for the MixpanelProvider.

### PR DESCRIPTION
**Pull request by ed-at-work**
This avoids using the singleton-style Mixpanel API.  I'm working on a project where we need more than one Mixpanel tracker available during the execution of our app.  Also this PR adds a new method to create an alias for a distinct id.

### CODE REVIEW

**Comment by orta** [comment]
Is this backwards compatible? 

In that, if I had code in the app that accesses the Mixpanel shared instance directly and only used one instance would the two still call methods on the same thing?

**Comment by ed-at-work** [comment]
No, I don't think it is.  One solution if needed might be to initialize the mix panel singleton instance the first time the ARAnalytics Mixpanel provider is used, and then subsequent ARAnalytics mix panel providers would store a separately created non-singleton mix panel instance.  Should I go ahead and give that a shot?

**Comment by orta** [comment]
Yeah, that makes sense :dancers:

**Comment by ed-at-work** [comment]
OK, I finally had time to add the couple of lines needed. This seems to work.  Do you want me to squash and rebase into a single commit, or is this fine?  Any other issues, please let me know.

File: Providers/MixpanelProvider.m [hunk 1]
```diff
@@ -14,10 +18,18 @@ - (id)initWithIdentifier:(NSString *)identifier andHost:(NSString *)host {
...
 
     NSAssert([Mixpanel class], @"Mixpanel is not included");
-    [Mixpanel sharedInstanceWithToken:identifier];
+
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _mixpanel = [Mixpanel sharedInstanceWithToken:identifier];
+    });
+
+    if(! _mixpanel) {
```
Comment (orta) [inline Providers/MixpanelProvider.m]: perfect :+1: - great usage of the dispatch once

**Comment by orta** [comment]
Re: rebase - nope, I'm not that fussy about history, and it's only two commits anyway. This looks great. Merging. Unless you're rushing for an official release, I'll wait until a 2-2 more PRs merge, and make a new build :+1: .

**Pull request merged**